docs: Add error pages for hydration related error codes (#49689)
angular/angular: aio/content/errors/NG0503.md

@name Hydration Unsupported Projection of DOM Nodes
@category runtime
@shortDescription During serialization, Angular detected DOM nodes that were created outside of Angular's context and used for content projection.

@description
This error means that during the server side serialization process, Angular encountered nodes that were created outside of Angular's context (i.e. using native DOM APIs) and found those nodes were provided as projectible nodes. They were likely provided using `ViewContainerRef.createComponent` or `createComponent` APIs. Angular hydration does not support this use case.

More information about hydration can be found in [this guide](guide/hydration).

The following examples will trigger the error.

```typescript
@Component({
  standalone: true,
  selector: 'dynamic',
  template: `

>   <ng-content />
>   <ng-content />

Reviewer: We can simplify to just one projection slot here.
```suggestion
  <ng-content />
```

`,
})
class DynamicComponent {
}

@Component({
  standalone: true,
  selector: 'app',

>   imports: [NgIf, NgFor],
>   template: `
>   <div #target></div>
>   <main>Hi! This is the main content.</main>
> `,

Reviewer: We can also simplify the code a bit:
```suggestion
  template: `<div #target></div>`,
```

})
class SimpleComponent {
  @ViewChild('target', {read: ViewContainerRef}) vcr!: ViewContainerRef;
  envInjector = inject(EnvironmentInjector);

  ngAfterViewInit() {
    const div = document.createElement('div');
    const p = document.createElement('p');

>     const span = document.createElement('span');
>     const b = document.createElement('b');
>     // In this test we create DOM nodes outside of Angular context
>     // (i.e. not using Angular APIs) and try to content-project them.
>     // This is an unsupported pattern and we expect an exception.
>     const compRef = createComponent(DynamicComponent, {
>       environmentInjector: this.envInjector,
>       projectableNodes: [[div, p], [span, b]]
>     });

Reviewer: We can drop "span" and "b" elements if we have 1 projection slot:
```suggestion
    // In this test we create DOM nodes outside of Angular context
    // (i.e. not using Angular APIs) and try to content-project them.
    // This is an unsupported pattern and an exception will be thrown.
    const compRef = createComponent(DynamicComponent, {
      environmentInjector: this.envInjector,
      projectableNodes: [[div, p]]
    });
```

    compRef.changeDetectorRef.detectChanges();
  }
}
```

@debugging

> Avoid direct DOM Manipulation using native APIs whenever possible.
> Add the `ngSkipHydration` attribute to your component host node.

Reviewer: (I've just did a copy'n'paste from the comment in NG0500)
```suggestion
The error message in the developer console should contain information on a specific part of the
application's DOM structure that is causing the problem. Review that part of the application
for hydration-related errors, such as direct DOM manipulation using native APIs.

You can add the `ngSkipHydration` attribute to your component host node as a possible workaround.
```
